Allow returning IActionResult and auto Complex Type serialize to Json

eByte23 opened the pull request:
This PR put support for return result that Inherit from IActionResult e.g. FileContentResult etc..
The dependency on  Mvc DI extension is for IActionExecutor which would be used for a lot of different ActionResult's

markvincze commented:
Hi!

Thanks for your PR, sorry for the delayed review.  
Actually I'v been thinking about how to enable using different result objects than just `string`s, and I might rework this part in the future, but I think this PR is a good first step.
Can you please write two new unit tests, one for testing the `IActionResult`-case, one for testing when the response is `object`?

markvincze commented on src/Stubbery/Stubbery.csproj:
```diff
@@ -3,7 +3,7 @@
     <Description>Library for creating Api stubs in .NET.</Description>
     <Copyright>Copyright © Mark Vincze 2018</Copyright>
-    <VersionPrefix>2.3.0</VersionPrefix>
+    <VersionPrefix>2.4.0</VersionPrefix>
```
I merged another PR in the meantime, can you bump this to 2.5.0?

markvincze commented on src/Stubbery/RequestMatching/SetupResponseParameters.cs:
```diff
@@ -28,7 +32,29 @@ public async Task SendResponseAsync(HttpContext httpContext)
-            await httpContext.Response.WriteAsync((string)Responder(httpContext.Request, arguments));
+            var response = Responder(httpContext.Request, arguments);
+
+            if (response is string)
```
In all 3 `if`s, please use this syntax:
```
            if (response is string stringResponse)
            {
                await httpContext.Response.WriteAsync(stringResponse);
                return;
            }
```
So that we're not casting twice.

eByte23 commented:
Thanks for the reply, I'll make the changes and some tests a little later today.

eByte23 commented:
@markvincze I have rebased the pr, resolved all comments and added the requested tests.
If there are any other changes you'd like let me know.
Cheers

markvincze commented:
@eByte23,

I'm afraid there is a problem now, in the new version we're not setting the status code, because this line:
```
httpContext.Response.StatusCode = StatusCodeProvider(httpContext.Request, arguments);
```
was removed, so we're returning `404` instead of `200`. You'll see if you run all the tests locally.  
I guess if you add that to the same place it was, that'll be the correct behavior, right? Because the ActionResult-branch possibly runs afterwards, and it can still change the status code to what it wants.

And unrelated issue is that we have a problem with the CI build, it doesn't really run the tests. I'll try to fix that separately on master.

eByte23 commented:
Sorry about that missed it in the rebase as it was the same line as the response.write before.

Should have revert that now, tests ran locally fine.

markvincze commented:
Great, thanks for the contribution!
I'll merge the PR now, and will release the new package a bit later, probably today.

markvincze commented:
Sorry for the delay, I released 2.5.0 now, it includes the PR.

eByte23 commented:
No problem at all much appreciated
